pkg/user/delivery/http/user_proxy.go

```diff
@@ -14,25 +14,40 @@ import (
 // UserProxy represents the interface to the user's proxy operations
 type UserProxy interface {
 	Create(user entities.User) (err error)
-	CreateToken(user entities.User) (string, error)
+	CreateUserToken(user entities.User) (token string, err error)
+	CreateAppToken(user entities.User, duration int) (token string, err error)
 }
 
 // Proxy is responsible for implementing the user's proxy operations
 type Proxy struct {
-	url    string
-	logger logging.Logger
+	userURL string
+	authURL string
+	logger  logging.Logger
 }
 
-// TokenResponse represents the create token response from the user's service
-type TokenResponse struct {
+// UserTokenResponse represents the create token response from the user's service
+type UserTokenResponse struct {
 	Token string `json:"token"`
 }
 
+// AppTokenRequest represent the create token request for the auth service
+type AppTokenRequest struct {
+	Email    string `json:"issuer"`
+	Type     int    `json:"type"`
+	Duration int    `json:"duration"`
+}
+
+// AppTokenResponse represents the create token response from the auth service
+type AppTokenResponse struct {
+	Token string `json:"value"`
+}
+
 // NewUserProxy creates a new Proxy instance
-func NewUserProxy(logger logging.Logger, hostname string, port uint16) *Proxy {
-	url := fmt.Sprintf("http://%s:%d", hostname, port)
-	logger.Debug("user proxy configured to " + url)
-	return &Proxy{url, logger}
+func NewUserProxy(logger logging.Logger, userHost string, userPort uint16, authHost string, authPort uint16) *Proxy {
+	userURL := fmt.Sprintf("http://%s:%d", userHost, userPort)
+	authURL := fmt.Sprintf("http://%s:%d", authHost, authPort)
```

Comment:
Since Authn is another service would not be correct to create an AuthnProxy?

Reply:
It's right! I gonna add a new file `authn_proxy.go` in the same directory

Reply:
Done!